src/v2/controllers/FormController.js

```diff
@@ -77,13 +77,29 @@ export default BaseLoginController.extend({
       return;
     }
 
+    this.toggleFormButtonState(true);
     model.trigger('request');
     return actionFn(model.toJSON())
       .then(resp => {
         this.options.appState.trigger('remediationSuccess', resp.response);
       })
       .catch(error => {
         model.trigger('error', model, {'responseJSON': error}, true);
+        this.toggleFormButtonState(false);
       });
   },
+
+  /**
+   * SignIn widget has its own (hacky) way to customize the button disabled state:
+   * adding `link-button-disabled` despite the name was intend only to disable
+   * `link-button`.
+   * Instead of doing decent refactor, we want to follow the convention for now.
+   *
+   * @param {boolean} disabled whether add extra disable CSS class.
+   */
+  toggleFormButtonState: function (disabled) {
+    var button = this.$el.find('.o-form-button-bar .button');
+    button.toggleClass('link-button-disabled', disabled);
```

Comment:
was this class not getting added to this button ?

Reply:
Courage doesn't add `link-button-disabled` for `input` buttons.